hs_composite_resource/tests/test_composite_resource.py

```diff
@@ -709,178 +800,496 @@ def test_supports_folder_creation(self):
+        tif_res_file = self.composite_resource.files.first()
+        GeoRasterLogicalFile.set_file_type(self.composite_resource, self.user, tif_res_file.id)
+        res_file = self.composite_resource.files.first()
+        self.assertNotEqual(res_file.file_folder, None)
+
+        # create a new folder
+        new_folder = '{}_1'.format(res_file.file_folder)
+        new_folder_path = os.path.join("data", "contents", new_folder)
+        create_folder(self.composite_resource.short_id, new_folder_path)
+
+        # test it should be possible to move the 'small_logan' folder that
+        # represents the raster aggregation to the 'my-new-folder' that we created above
+        src_full_path = os.path.join(self.composite_resource.file_path, res_file.file_folder)
+        tgt_full_path = os.path.join(self.composite_resource.file_path, new_folder)
+        # this is the function we are testing
+        self.assertEqual(self.composite_resource.supports_rename_path(
+            src_full_path, tgt_full_path), True)
+
+        self.composite_resource.delete()
+
+    def test_supports_zip_non_aggregation_folder(self):
+        """Here we are testing the function supports_zip()
+        Test that zipping a folder that contains file(s) that is not part of any aggregation
+        is allowed
+        """
+
+        self.create_composite_resource()
+
+        # add a file to the resource
+        self.add_file_to_resource(file_to_add=self.generic_file)
+        self.assertEqual(self.composite_resource.files.count(), 1)
+        new_folder = 'my-new-folder'
+        new_folder_path = "data/contents/{}".format(new_folder)
+
+        # create a new folder
         create_folder(self.composite_resource.short_id, new_folder_path)
         # now move the file to this new folder
+        src_path = os.path.join('data', 'contents', self.generic_file_name)
+        tgt_path = os.path.join(new_folder_path, self.generic_file_name)
         move_or_rename_file_or_folder(self.user, self.composite_resource.short_id,
-                                      'data/contents/' + self.generic_file_name,
-                                      new_folder_path + "/" + self.generic_file_name)
-        folder_to_zip = self.composite_resource.short_id + '/data/contents/my-new-folder'
-        # test that we can zip the folder my_new_folder
+                                      src_path, tgt_path)
+
+        # test that we can zip the folder my_new_folder as this folder has no aggregation
+        folder_to_zip = os.path.join(self.composite_resource.file_path, new_folder)
         self.assertEqual(self.composite_resource.supports_zip(folder_to_zip), True)
 
-        # test that a folder containing resource files that are part of the GeorasterLogicalFile
-        # can be zipped
-        self._add_raster_file_to_resource()
-        self.assertEqual(self.composite_resource.files.count(), 2)
+        self.composite_resource.delete()
```

Comment:
These tests are very thorough 👍 
I think we should probably put resource deletion in the teardown method.  It helps group similar tests and also guarantees the resource will be deleted even if an exception is thrown in the test run.

Reply:
Probably should use setup as well for the resource creation?

Reply:
Will add tearDown() for resource delete